pandas/_libs/reduction.pyx

```diff
@@ -203,7 +203,8 @@ cdef class SeriesBinGrouper:
         self.f = f
 
         values = series.values
-        if not values.flags.c_contiguous:
+        if util.is_array(values) and not values.flags.c_contiguous:
+            # e.g. Categorical has no `flags` attribute
```

Comment:
does this work with Categorical otherwise (the BinGrouper)? do we need tests here?

Reply:
we have tests that get here with Categorical and IntegerArray.